Feature/accounts resolver

WanjikuMac opened the pull request:
Fixes #7

WanjikuMac committed:
added scopes that may use custom stacks

WanjikuMac committed:
plug to excecute context requests

WanjikuMac committed:
The GraphQl schema platform

WanjikuMac committed:
middleware authentication

WanjikuMac committed:
helper functions to help format changeset errors

WanjikuMac committed:
account types

zacck-zz commented on lib/hotel_web/schema/middleware/changeset_errors.ex:
```diff
@@ -0,0 +1,33 @@
+    @spec build_error(tuple()) :: String.t()
+    defp build_error({msg, opts}) do
+      Enum.reduce(opts, msg, fn({key, value}, acc) ->
+        String.replace(acc, "%{#{key}}", to_string(value))
```
@WanjikuMac  this error could include more details usually in an `opts` field

WanjikuMac replied:
@zacck Could you please elaborate or do you mean something like this? And is it implemented in the `@behavoir` functions of `Absinthe.Middleware`
```defp build_error({msg, opts(attrs)}) do```

zacck-zz replied:
I mean the `opts` item has more details usually about the error that the user can use to recover

WanjikuMac replied:
@zacck Oh okay, how would I then can I added a detailed error msg with even a couple of solutions?

zacck-zz replied:
like this https://github.com/LittleKidogo/quick_bytes_backend/blob/3f8c1d75b26c5835f70eed3a464351e806a104d9/lib/qb_backend_web/schema/middleware/changeset_errors.ex#L30

WanjikuMac replied:
@zacck Which is the same thing I have ama?

WanjikuMac committed:
added the accounts resolvers

WanjikuMac committed:
Log in reolver test

WanjikuMac committed:
renamed Hotel.Router.Helper to HotelWeb.Router.Helper

 added the guardian set up

rename the atom :Hotel to :hotel

changed the scope

renamed sign in params to sign in param

renamed sign in params to sign in param for account test

WanjikuMac committed:
added usr sign up and fetch users tests

WanjikuMac commented:
@zacck final tests for this PR, your review will be highly appreciated

zacck-zz commented:
@WanjikuMac  looks awesome, coming well keep going :)
